ztp/gitops-subscriptions/ACMPolicyGeneratorIntergration.md

```diff
@@ -0,0 +1,183 @@
+  manifests:
+    - path: custom_source_crs/validationCRs/
+```
+
+The annotation "ran.openshift.io/ztp-deploy-wave" is required to be added in the policy for [TALM](https://github.com/openshift-kni/cluster-group-upgrades-operator) to know the order of policies to be applied/checked. This can be done via `policyDefaults.policyAnnotations` as shown above.
+
+* Add `extended-du-validator.yaml` in the `kustomization.yaml` file as a generator.
+
+
+See [ACM PolicyGenerator reference YAML](https://github.com/stolostron/policy-generator-plugin/blob/main/docs/policygenerator-reference.yaml) for the full structure of `PolicyGenerator`. 
+
+### Integrating with Openshift GitOps(ArgoCD)
+As ACM Policy Generator is not preinstalled in the Openshift GitOps container image, these steps describes how to do integrate it with Openshift GitOps.
+
+**Requirements**:
+1. Advanced Cluster Management(ACM) operator v2.4+ installed on the hub cluster
+1. Redhat Openshift GitOps operator v1.4/1.5 installed and patched with [ZTP container](argocd/deployment/argocd-openshift-gitops-patch.json) on the hub cluster
+
+**Steps**:
+1. Patch the ArgoCD to copy the policy generator binary from the ACM application subscription container image to the ArgoCD container.
+* Get the current `argocd` object
+```
+oc get argocd -n openshift-gitops openshift-gitops -o json > openshift-gitops-argocd.json 
+```
+* Edit the downloaded `openshift-gitops-argocd.json`file with the ACM subscription image added in the `initContainers` like in the example.
+```
+{
+  "spec": {
+    "repo": {
+      "initContainers": [
+        ...
+        {
+          "name": "acm-policy-generator",
+          "command": [
+            "/bin/bash"
+          ],
+          "args": [
+            "-c",
+            "cp -r /etc/kustomize /.config"
+          ],
+          "imagePullPolicy": "Always",
+          "volumeMounts": [
+            {
+              "name": "kustomize",
+              "mountPath": "/.config"
+            }
+          ],
+          "image": "quay.io/stolostron/multicluster-operators-subscription:2.5.0"
+        }
+      ]
+    }
+  }
+}
+```
+* Apply the patch
+```
+oc patch argocd openshift-gitops -n openshift-gitops --type=merge --patch-file openshift-gitops-argocd.json
+```
+
+2. Update the existing `AppProject` for policy app created via [policy-app-project.yaml](argocd/deployment/policies-app-project.yaml) with the `PolicyGenerator` object.
+```
+oc edit appprojects.argoproj.io -n openshift-gitops policy-app-project
+
+spec:
+  namespaceResourceWhitelist:
+  - group: policy.open-cluster-management.io/v1
+    kind: PolicyGenerator
+```
+
+3. Push the prepared custom source files and `PolicyGenerator` yamls to the Git respository, ArgoCD will detect the changes and generate the policies.
+Note: you could push to the same Git directory as the current Policy app references to or a new directory and create a new Argo application with that directory to roll out the changes.
```

Comment:
What is being referred to by "current Policy app references". Does this mean the current PolicyGenTemplate yaml files handled by the ztp-site-generator version of the plugin? If so, maybe clarify?